### PR TITLE
Remove usage of simple history

### PR DESCRIPTION
**Pull request by ahsan-ul-haq**
LEARNER-3348

### CODE REVIEW

File: ecommerce/courses/models.py [hunk 1]
```diff
@@ -39,7 +38,14 @@ class Course(models.Model):
         blank=True,
         help_text=_('Last date/time on which verification for this product can be submitted.')
     )
-    history = HistoricalRecords()
+    created = models.DateTimeField(
```
Comment (clintonb) [inline ecommerce/courses/models.py]: Don't manually make these fields. Use `from django_extensions.db.models import TimeStampedModel`. Also, make sure these fields are populated with the data from the first and last historical record.

**Review by clintonb — CHANGES_REQUESTED** [review]

File: ecommerce/courses/models.py (hunk 1 above)
Comment (ahsan-ul-haq) [reply ecommerce/courses/models.py]: I tried using that but it requires a default value as, I believe, it does not allow null=True.

Comment (clintonb) [reply ecommerce/courses/models.py]: Drop the fields.

Comment (ahsan-ul-haq) [reply ecommerce/courses/models.py]: So we are not going to need that as well?

Comment (clintonb) [reply ecommerce/courses/models.py]: Sorry. Keep your customizations, and populate with the historical records.

**Comment by ahsan-ul-haq** [comment]
@clintonb madee the change

**Comment by ahsan-ul-haq** [comment]
@clintonb please take a look.

File: ecommerce/courses/migrations/0007_auto_20171205_0800.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
```
Comment (clintonb) [inline ecommerce/courses/migrations/0007_auto_20171205_0800.py]: Merge this with the migration above.

File: ecommerce/courses/migrations/0007_auto_20171205_0800.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+# Generated by Django 1.10.7 on 2017-12-05 08:00
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+from ecommerce.core.constants import ISO_8601_FORMAT
+from ecommerce.courses.models import Course
```
Comment (clintonb) [inline ecommerce/courses/migrations/0007_auto_20171205_0800.py]: This is not the proper way to access models in migrations. Use `apps.get_models()`. See https://docs.djangoproject.com/en/1.11/topics/migrations/#data-migrations.

File: ecommerce/courses/migrations/0007_auto_20171205_0800.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        for course in courses:
+            course.modified = None
+            course.save()
+
+    dependencies = [
+        ('courses', '0006_auto_20171204_1036'),
+    ]
+
+    operations = [
+        migrations.RunPython(add_created_date, remove_created_date),
```
Comment (clintonb) [inline ecommerce/courses/migrations/0007_auto_20171205_0800.py]: Merge these into a single method, and avoid iterating over the data twice.

File: ecommerce/courses/migrations/0007_auto_20171205_0800.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+from ecommerce.courses.models import Course
+
+
+class Migration(migrations.Migration):
+
+    def add_created_date(apps, schema_editor):
+        courses = Course.objects.all()
+
+        for course in courses:
+            created_date = course.history.earliest().history_date.strftime(ISO_8601_FORMAT)
```
Comment (clintonb) [inline ecommerce/courses/migrations/0007_auto_20171205_0800.py]: Why are you converting the `datetime` object to a string? That is not necessary.

File: ecommerce/courses/migrations/0007_auto_20171205_0800.py (hunk 4 above)
Comment (clintonb) [reply ecommerce/courses/migrations/0007_auto_20171205_0800.py]: Also, don't bother with a revert. If we revert the merged migration, the column will be deleted. Use [`RunPython.noop`](https://docs.djangoproject.com/en/1.11/ref/migration-operations/#django.db.migrations.operations.RunPython.noop).

File: ecommerce/courses/models.py (hunk 1 above)
Comment (clintonb) [reply ecommerce/courses/models.py]: Nit: Both of these declarations can be on one line.

File: ecommerce/courses/models.py [hunk 6]
```diff
@@ -39,8 +39,16 @@ class Course(models.Model):
...
+        null=True,
+        auto_now_add=True
+    )
+    modified = models.DateTimeField(
+        null=True,
+        auto_now=True
+    )
     thumbnail_url = models.URLField(null=True, blank=True)
+    history = HistoricalRecords()
```
Comment (clintonb) [inline ecommerce/courses/models.py]: Nit: put this back where it was.

File: ecommerce/extensions/api/serializers.py [hunk 7]
```diff
@@ -298,10 +297,7 @@ def __init__(self, *args, **kwargs):
             self.fields.pop('products', None)
 
     def get_last_edited(self, obj):
-        try:
-            return obj.history.latest().history_date.strftime(ISO_8601_FORMAT)
-        except ObjectDoesNotExist:
-            return None
+        return obj.modified.strftime(ISO_8601_FORMAT)
```
Comment (clintonb) [inline ecommerce/extensions/api/serializers.py]: What happens if the value is `None`?

File: ecommerce/courses/models.py [hunk 8]
```diff
@@ -39,8 +39,16 @@ class Course(models.Model):
         blank=True,
         help_text=_('Last date/time on which verification for this product can be submitted.')
     )
-    history = HistoricalRecords()
+    created = models.DateTimeField(
+        null=True,
+        auto_now_add=True
+    )
+    modified = models.DateTimeField(
+        null=True,
```
Comment (clintonb) [inline ecommerce/courses/models.py]: Make these fields non-nullable in a follow-on PR.

**Review by clintonb — COMMENTED** [review]
Fix the migration, and update the translations.

**Comment by ahsan-ul-haq** [comment]
@clintonb made the suggested changes.

File: ecommerce/courses/migrations/0006_auto_20171204_1036.py [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+    def add_created_modified_date(apps, schema_editor):
+        Course = apps.get_model('courses', 'Course')
+        courses = Course.objects.all()
+
+        for course in courses:
+            created_date = course.history.earliest().history_date
```
Comment (waheedahmed) [inline ecommerce/courses/migrations/0006_auto_20171204_1036.py]: No need to create variables, just directly assign the value to model fields.

**Review by waheedahmed — APPROVED** [review]
Just one comment otherwise :+1:

**Comment by ahsan-ul-haq** [comment]
@waheedahmed made the change. @clintonb please review

**Review by clintonb — APPROVED** [review]
Please ensure the [Hawthorn release notes](https://openedx.atlassian.net/wiki/spaces/COMM/pages/158045104/Hawthorn) are updated to inform maintainers that django-simple-history is being removed from this service.

**Comment by clintonb** [comment]
FYI @edx/learner @edx/devops @edx/open-source-team

**Comment by ahsan-ul-haq** [comment]
Closing this and creating a new PR https://github.com/edx/ecommerce/pull/1587

**Pull request closed without merging**